### PR TITLE
Add support for SendGrid's asm_group_id

### PR DESCRIPTION
**Pull request by richeterre**
This is useful for letting recipients unsubscribe from certain types of email communication. By assigning an ASM group id to the email, hitting "Unsubscribe" in the footer will unsubscribe the recipient only from this email group, not all emails.

### CODE REVIEW

**Comment by richeterre** [comment]
I've cleaned up the "conflict" (it was just some indenting issue), so maintainers are welcome to merge this if they so wish!

**Comment by tomtaylor** [comment]
It'd be great to get this merged! Can I help?

File: lib/bamboo/adapters/send_grid_helper.ex [hunk 1]
```diff
@@ -83,6 +84,23 @@ defmodule Bamboo.SendGridHelper do
     raise "expected a list of category strings"
   end
 
+  @doc """
+  An integer id for an ASM (Advanced Suppression Manager) group that this email should belong to.
+  This can be used to let receipients unsubscribe from only a certain type of communication.
```
Comment (maymillerricci) [inline lib/bamboo/adapters/send_grid_helper.ex]: "receipients" -> "recipients"

**Comment by tomtaylor** [comment]
To help getting this merged I've made the changes suggested in a new pull request: #457.

**Comment by maymillerricci** [comment]
Closing this since the changes were incorporated in #457. Thanks for your work on this!

**Pull request closed without merging**